### ISSUE TITLE
run host's apps from within sanbox - the options?

### ISSUE THREAD

**Issue by bam80** [flatpak/flatpak#2436]
Consider Discover flatpak: https://phabricator.kde.org/source/flatpak-kde-applications/browse/master/org.kde.discover.json
As an App Center, it already can list, "dicover", install and remove apps on a host, which is pretty useful.
But, it still lack possibility to launch the apps, which makes it not fully-usable for user.
What can be an options to add such capability to it?

**Comment by TingPing** [comment]
`--talk-name=org.freedesktop.Flatpak`, `flatpak-spawn --host any-command`.

Though personally I believe some applications just aren't meant to be Flatpaked and that includes software managers.

**Comment by bam80** [comment]
I'm wondering if `xdg-desktop-portal` can take that role, without relying on `flatpak-spawn`?
It already has interfaces for file access, opening URIs, printing and others. Doesn't running the programs seems reasonable addition? After all, it's all XDG standard things..

**Comment by TingPing** [comment]
> Doesn't running the programs seems reasonable addition? 

No, its a sandbox, executing anything on the host is an escape.

**Comment by bam80** [comment]
But when you accessing the file, opening URIs, printing and others, you already execute something on the host anyway. Doesn't it all what `xdg-desktop-portal` needed for, considered it all possible only if enabled in the manifest?

**Comment by matthiasclasen** [comment]
portals don't need to be enabled in the manifest, they are available to all sandboxed apps for free, since they are supposed to be secure and keep the user in control. flatpak-spawn talks to a dbus service as well, but its not a portal, so it needs to be explicitly allowed in the manifest.

Why OpenURI is a portal and Spawn isn't can certainly be debated. It is a matter of degrees of control - explicitly specifying an arbitrary command to run gives a lot more control and attack surface than asking to open a file.

I agree with TingPing that it seems pretty hopeless to use sandboxing for deeply system-intertwined things like file managers, software installers or debugging tools. That doesn't mean that you can't use flatpak as a deployment and distribution mechanism, of course. But you'll have to be prepared to poke holes into the sandbox for the things you need.

**Comment by bam80** [comment]
Thanks, I'll try to punch a hole in the sanbox then, maybe @aleixpol wouldn't mind..
About using sandboxing for system software, I have to do it for personal reason - my distro keeper denied to package Discovery, and I wouldn't change the distro nor to package the software myself.
Meanwhile, Discovery flatpak already showed remarkable results, that is why the question occurred.
